File: Array/74-search-a-2d-matrix/search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int rows=matrix.size();
        int cols=matrix[0].size();
        int low=0;
        int high=rows*cols-1;
        while(low<=high){
            int mid=low+(high-low)/2;
            int row=mid/cols;
            int col=mid%cols;
            if(matrix[row][col]==target){
                return true;
            }
            else if(matrix[row][col]>target){
                high=mid-1;
            }
            else{
                low=mid+1;
            }
        }
        return false;
    }
};

File: Binary_Tree/74-search-a-2d-matrix/search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {

        int n = matrix.length;
        int n1= matrix[0].length;

        int start=0;
        int end = n*n1-1;

        while(start<=end){

            int mid = start+(end-start)/2;

            int row = mid/n1;
            int col = mid%n1;

            if(matrix[row][col]==target) return true;

            else if(matrix[row][col]>target){
                end= mid-1;
            }
            else{
                start= mid+1;
            }
        }
        return false;
        
    }
}